src/WebAppDIRAC/WebApp/handler/ResourceSummaryHandler.py

```diff
@@ -148,24 +157,18 @@ def setToken(self, requestParams):
         )
 
         if not res["OK"]:
-            self.finish({"success": "false", "error": res["Message"]})
-        else:
-            self.finish({"success": "true", "result": res["Value"]})
+            return {"success": "false", "error": res["Message"]}
+        return {"success": "true", "result": res["Value"]}
 
-    def setStatus(self, requestParams):
-        username = self.getUserName()
+    def setStatus(self, requestParams: dict) -> dict:
+        """Set token
 
-        if username == "anonymous":
-            self.finish({"success": "false", "error": "Cannot perform this operation as anonymous"})
-            return
-        if "SiteManager" not in self.getProperties():
-            self.finish({"success": "false", "error": "Not authorized"})
-            return
-
-        pub = PublisherClient()
+        :param requestParams: request parameters
+        """
+        if error := self.__checkAuth(username := self.getUserName()):
+            return {"success": "false", "error": error}
 
-        res = yield self.threadTask(
-            pub.setStatus,
+        res = PublisherClient().setStatus(
```

Comment:
What was wrong with the previous implementation?

Reply:
This place in the code already in a second (non-main) thread.

```python
class ResourceSummaryHandler(SummaryHandlerMix):
    ...
    @asyncGen
    def web_action(self):
        callback = yield self.threadTask(self._action)  # this line run all sync logic in a second thread
        self.finish(callback)
```
`web_action` run in thread `_action` that run `setStatus` that run `PublisherClient().setStatus(..`